src/pages/creations/LogosCarousel/index.tsx

```diff
@@ -0,0 +1,301 @@
+import { styled } from 'styled-components';
+import { MOBILE_MAX_WIDTH, TABLET_MAX_WIDTH } from '~/components';
+import { Carousel } from 'react-responsive-carousel';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { isMobile, isTablet } from '~/utils';
+
+interface LogosCarouselProps {
+  companies: string[];
+  onChange?: (index: number) => void;
+}
+
+const carouselState = {
+  desktop: {
+    loop: 7,
+    visibleItems: 9,
+  },
+  tablet: {
+    loop: 5,
+    visibleItems: 5,
+  },
+  mobile: {
+    loop: 5,
+    visibleItems: 3,
+  },
+};
+
+const formatCompanyToFileName = (company: string) => company.toLowerCase().replace(/ /g, '-');
+
+export default function LogosCarousel({ companies, onChange }: LogosCarouselProps) {
+  const [loop, setLoop] = useState(carouselState.desktop.loop); // iterations of the logos
+  const [visibleItems, setVisibleItems] = useState(carouselState.desktop.visibleItems); // number of logos visible
+
+  const maxSize = useMemo(() => Math.ceil(visibleItems / 2), [visibleItems]); // max size of the logos
+  const centerIndex = useMemo(() => Math.floor(visibleItems / 2), [visibleItems]); // index of the center logo
+  const initialIndex = useMemo(() => visibleItems * Math.floor(loop / 2), [visibleItems, loop]); // start in the middle of the list, from index 0
+
+  const items = useMemo(() => Array(loop).fill(companies).flat(), [loop, companies]); // repeat the companies to fill the loop
+  const [selectedItem, setSelectedItem] = useState(initialIndex);
+
+  // setSize is used to determine the size of the logos
+  // if the index is the selected item, the size is the max size, Math.ceil(VISIBLE_ITEMS / 2)
+  // otherwise, if the index is bigger or smaller than the selected item, the size is the module of the difference between selected item and index
+  const setSize = (index: number, currIndex: number) => {
```

Comment:
This function could be renamed to setLogoSizeByIndex and avoid extra comments, the code explains itself.

Reply:
https://github.com/defi-wonderland/web/pull/114/commits/bdf96f491f213c63ab3ba85c7e8f07cfdff5f739